### DISCUSSION TITLE
Weird problem with squared machine

### DISCUSSION THREAD

**Discussion by HalbesHaehnchen** [phil-barrett/grblHAL-teensy-4.x#98] [Q&A]
Hello again,

looking for a hint what could be wrong here. I set up a new machine and had problems with unclosed shapes (circle lines dont hit each other). Since its the 3rd machine I built, I already knew its the v-slot wheels that sit to tight.  I installed new ones and got rid of that problem. 
However, more complicated shapes are still broken. I thought its the same issue but its not. Machine frame is square, belt is fine, nothing is screwed together to tight or lose. When I move the Y gantry by hand it hits the frame at the same time on both sides. 

Now when I set an offset for the auto square feature it suddenly gets better. Now the gantry has a little gap when I send it on the very end of the workspace but the engraving is much more accurate. 

What could cause this? Its probably a physical problem, but where? Since I checked (and disassembled) almost everything multiple times I am a bit lost now.  
Maybe someone has an Idea I didnt think of yet.
![1702681920978 6497](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/19913295/7f34b263-925f-4dc2-acca-eed812fc3d79)

**Comment by phil-barrett** [comment]
Have you read my [blog on autosquare](https://www.grbl.org/single-post/it-s-hip-to-be-auto-square)?   It has a process for determining where your limit switches should be set to have a square gantry.  If your limit switch targets are not adjustable, you could adapt the procedure to find the correct Y dual axis offset ($171).  The basic idea is to cut 4 holes (or any feature) in a rectangular pattern (or burn a rectangle with a laser) and measure the diagonals.  When they are equal, your gantry is square. Knowing how far out of square it is, you could probably come up with a fairly good number to change $171 by.

Hmmm, I feel a new blog coming on...

Edit:  I realize I did not directly answer your question but I suspect your frame is not truly square. It appears that with $171=0, there is a serious misalignment.  Try burning a rectangle and measuring the corners.  It is also possible that you have some backlash in your machine.

**Reply by HalbesHaehnchen** [reply]
Hi :) Sure I read your blog. The problem is not about auto squaring, I just mentioned it because with an offset it works much better and I dont get it why.  The frame is square, I really paid attention to that, the table the machine is installed on is very even/flat. 
I did a squaring test anyway and it came out as expected, its squared. 

Backlash should be reduced by using very low feedrates right? Doesnt make a diffrence in my case if I do it with 900mm/m or 100.
Only last thing I could think of is its the "Everman Belt Drive System" I used. It works like rack/pinion with a second belt glued into the v-slot profile. I could easily remove it for testing but it would be a massive pain to reinstall. Thats why I didnt check that yet. 
I dont see why it should affect the accuracy that much or why an $171 offset makes it better.

![Unbenannt-1](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/19913295/9ce5d271-bfd8-499b-848e-2feceb712884)

**Reply by phil-barrett** [reply]
When you talked about using $171, I assumed you were dealing with gantry squareness issues. Using it causes the gantry end (second drive side, I believe) to be pull to a position relative to the end-stop sensor (aka limit switch).

I don't see how a low feedrate would compensate for backlash. You can test to see if your machine has backlash. In the laser case, I would do a "staircase" test, see below.  There is backlash compensation built into grblHAL. You will need to build new firmware. Enable it on the Advance features tab of the Web Builder.  Note - BL testing and compensation should only be applied after you have tightened up your mechanical components.

![image](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/16570958/20357674-ce1c-4922-9879-72657481f852)
Make sure the machine engraves in the direction of the arrows.  If the strokes are of a different length, the machine has backlash.

**Reply by HalbesHaehnchen** [reply]
Did that but without clear results. Pretty hard to measure. Anyway, I solved it. 
It was the "double belt" system I used. It must have something to do with the tandem Y axis. In my mind, it should not really matter if both belts are 100% aligned to each other since the steppers use a high amount of microsteps but I guess it does matter. Or whatever else the reason was, it is solved now. 
On my X axis that system works just fine though. 

Thank you for the help, I know it has nothing to do with your controller and my posting didnt really fit here :)

**Reply by phil-barrett** [reply]
No problem, glad to have your questions here.

**Reply by HalbesHaehnchen** [reply]
maybe other little question, after turning the controller off, I can't home it with my software. I have to use ioSender to unlock the machine. $x wont do it, I always get error:79.  
Is it set by $484 ? If so, it wont let me change to 0. It switches back to 1 everytime.

**Reply by phil-barrett** [reply]
Not sure I understand, is "turning the controller off" a typo?  In ioSender you do a reset and an unlock.

**Reply by HalbesHaehnchen** [reply]
nah probably just my broken English knowledge. I mean when I disconnect the controller from (USB) power and it boots again after connecting it, it does not accept any commands. It probably boots into some Alarm status. I have to do the reset/unlock thing with ioSender. Is there a way to disable it and let the board just boot up unlocked ?

**Reply by phil-barrett** [reply]
I think the ioSender console will show you what it does. You will need to check the verbose box and uncheck the other two boxes.
